src/net/sourceforge/kolmafia/request/CharPaneRequest.java

```diff
@@ -1764,51 +1742,48 @@ public static final void parseStatus(final JSONObject json) throws JSONException
 
     CharPaneRequest.setInteraction();
 
-    if (KoLCharacter.getLimitMode().limitFamiliars()) {
-      // No familiar
-    } else if (KoLCharacter.inAxecore()) {
-      int level = json.getIntValue("clancy_level");
-      int itype = json.getIntValue("clancy_instrument");
-      boolean att = json.getBoolean("clancy_wantsattention");
-      AdventureResult instrument =
-          itype == 1
-              ? CharPaneRequest.SACKBUT
-              : itype == 2 ? CharPaneRequest.CRUMHORN : itype == 3 ? CharPaneRequest.LUTE : null;
-      KoLCharacter.setClancy(level, instrument, att);
-    } else if (KoLCharacter.isJarlsberg()) {
-      if (json.containsKey("jarlcompanion")) {
-        int companion = json.getIntValue("jarlcompanion");
-        switch (companion) {
-          case 1 -> KoLCharacter.setCompanion(Companion.EGGMAN);
-          case 2 -> KoLCharacter.setCompanion(Companion.RADISH);
-          case 3 -> KoLCharacter.setCompanion(Companion.HIPPO);
-          case 4 -> KoLCharacter.setCompanion(Companion.CREAM);
+    if (!KoLCharacter.getLimitMode().limitFamiliars()) {
+      if (KoLCharacter.inAxecore()) {
+        int level = json.getIntValue("clancy_level");
+        int itype = json.getIntValue("clancy_instrument");
+        boolean att = json.getBoolean("clancy_wantsattention");
+        AdventureResult instrument =
+            itype == 1
+                ? CharPaneRequest.SACKBUT
+                : itype == 2 ? CharPaneRequest.CRUMHORN : itype == 3 ? CharPaneRequest.LUTE : null;
+        KoLCharacter.setClancy(level, instrument, att);
+      } else if (KoLCharacter.isJarlsberg()) {
+        if (json.containsKey("jarlcompanion")) {
+          int companion = json.getIntValue("jarlcompanion");
+          switch (companion) {
+            case 1 -> KoLCharacter.setCompanion(Companion.EGGMAN);
+            case 2 -> KoLCharacter.setCompanion(Companion.RADISH);
+            case 3 -> KoLCharacter.setCompanion(Companion.HIPPO);
+            case 4 -> KoLCharacter.setCompanion(Companion.CREAM);
+          }
+        } else {
+          KoLCharacter.setCompanion(null);
+        }
+      } else if (KoLCharacter.getPath().canUseFamiliars()) {
+        int famId = json.getIntValue("familiar");
+        int famExp = json.getIntValue("familiarexp");
+        FamiliarData familiar = FamiliarData.registerFamiliar(famId, famExp);
+        KoLCharacter.setFamiliar(familiar);
+
+        String image = json.getString("familiarpic");
+        if (famId != FamiliarPool.MELODRAMEDARY) {
+          KoLCharacter.setFamiliarImage(image.isEmpty() ? null : image + ".gif");
         }
-      } else {
-        KoLCharacter.setCompanion(null);
-      }
-    } else if (KoLCharacter.isEd()) {
```

Comment:
again, this blank if skips the remaining else-if checks. Less important here because there's no else block.